Tests/BSONTests/BSONDocumentIteratorTests.swift

```diff
@@ -0,0 +1,28 @@
+@testable import BSON
+import Foundation
+import Nimble
+import XCTest
+
+final class DocumentIteratorTests: BSONTestCase {
+    func testFindByteRangeEmpty() {
+        let d: BSONDocument = [:]
+        var iter = d.makeIterator()
+        let range = iter.findByteRange(for: "item")
+        expect(range).to(beNil())
+    }
+
+    func testFindByteRangeItemsInt32() {
+        let d: BSONDocument = ["item0": .int32(32), "item1": .int32(32)]
+        var iter = d.makeIterator()
+        let range = iter.findByteRange(for: "item1")!
```

Comment:
If this unwrap fails, it will crash the entire test runner. I think it would be better to first verify that it is not nil via an `expect(...).toNot(beNil())` or via a `guard let ... else { XCTFail("reason") }` to prevent that.